src/main/java/io/github/wimdeblauwe/errorhandlingspringbootstarter/mapper/ErrorMessageMapper.java

```diff
@@ -12,13 +12,27 @@ public ErrorMessageMapper(ErrorHandlingProperties properties) {
         this.properties = properties;
     }
 
-    public String getErrorMessage(Throwable throwable) {
-        String exceptionClassName = throwable.getClass().getName();
+    public String getErrorMessage(Throwable exception) {
+         // Find the first existing message override throw the class hierarchy
+         String code = getErrorMessage(exception.getClass());
+        if (code != null) {
+            return code;
+        }
+        // If not found return the exception mesage
+        return exception.getMessage();
+    }
+
+    public String getErrorMessage(Class<?> exceptionClass) {
```

Comment:
1) Method can be private
2) Move private methods to end of the file
3) Use `Class<? extends Throwable>`